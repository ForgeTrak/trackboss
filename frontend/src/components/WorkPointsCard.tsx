import React from 'react';
import GaugeChart from 'react-gauge-chart';
import { Box, Center, Text, VStack } from '@chakra-ui/react';
import { useNavigate } from 'react-router-dom';

interface cardProps {
    percent: number,
}

export default function WorkPointsCard(props: cardProps) {
    const navigate = useNavigate();
    const navigateToWorkPoints = () => {
        const path = '/settings';
        navigate(path);
    };

    return (
        <Box
            onClick={navigateToWorkPoints}
            cursor="pointer"
            bg="white"
            boxShadow="md"
            border="1px"
            borderColor="gray.200"
            h="365"
            p={3}
            m={2}
        >
            <Center>
                <VStack p={0} boxSize="md">
                    <GaugeChart
                        hideText
                        id="gauge-chart"
                        nrOfLevels={20}
                        percent={props.percent / 100}
                        colors={['#FF9F46', '#76CE6F']}
                    />
                    <VStack spacing={0}>
                        <Text pt={5} fontSize="2xl">You have completed</Text>
                        <Text color="orange" fontSize="4xl">
                            {props.percent}
                            %
                        </Text>
                        <Text fontSize="2xl">of this year&apos;s work points</Text>
                    </VStack>
                </VStack>
            </Center>
        </Box>
    );
}
