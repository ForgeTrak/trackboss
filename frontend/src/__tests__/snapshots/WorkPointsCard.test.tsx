import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import WorkPointsCard from '../../components/WorkPointsCard';

jest.mock('react-gauge-chart', () => {
    return {
      'default': 'GaugeChart'
    }
  });


it('renders correctly', () => {
    const workPointsCard = renderer.create(
            <WorkPointsCard percent={67} />
    ).toJSON();
    expect(workPointsCard).toMatchSnapshot();
});