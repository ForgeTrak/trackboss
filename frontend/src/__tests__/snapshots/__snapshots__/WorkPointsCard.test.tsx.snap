// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="css-merm6w"
>
  <div
    className="css-gmuwbf"
  >
    <div
      className="chakra-stack css-1ak7tnr"
    >
      <img />
      <div
        className="chakra-stack css-jp20jv"
      >
        <p
          className="chakra-text css-12gxl7b"
        >
          You have completed
        </p>
        <p
          className="chakra-text css-ev573i"
        >
          67
          %
        </p>
        <p
          className="chakra-text css-1gjuw9l"
        >
          of this year's work points
        </p>
      </div>
    </div>
  </div>
</div>
`;
